// ionic4-rating/__ivy_ngcc__/dist/ionic-rating.module.js
import * as ɵngcc0 from '@angular/core';
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
import { NgModule } from "@angular/core";
import { FormsModule } from "@angular/forms";
import { IonicModule } from "@ionic/angular";
import { CommonModule } from '@angular/common';
import { RatingComponent } from "./components/rating";
var IonicRatingModule = /** @class */ (function () {
    function IonicRatingModule() {
    }
IonicRatingModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: IonicRatingModule });
IonicRatingModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function IonicRatingModule_Factory(t) { return new (t || IonicRatingModule)(); }, imports: [[IonicModule, CommonModule, FormsModule]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(IonicRatingModule, { declarations: function () { return [RatingComponent]; }, imports: function () { return [IonicModule, CommonModule, FormsModule]; }, exports: function () { return [RatingComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(IonicRatingModule, [{
        type: NgModule,
        args: [{
                imports: [IonicModule, CommonModule, FormsModule],
                exports: [RatingComponent],
                declarations: [RatingComponent]
            }]
    }], function () { return []; }, null); })();
    return IonicRatingModule;
}());
export { IonicRatingModule };
